src/App.jsx

```diff
@@ -14,17 +14,24 @@ function App() {
         "Lesson 1.5 Side effects, custom hooks, fragments",
     ]
     const initialBlogs = initialTitles.map(title => titleToBlog(title))
-    const [ blogTitles, setBlogTitles]= useSemiPersistentState('savedBlogTitles', initialBlogs)
+    const [blogTitles, setBlogTitles] = useSemiPersistentState('savedBlogTitles', initialBlogs)
     const [searchText, setSearchText] = react.useState('');
+    const [focus, setFocus] = react.useState('search');
 
+    // focus needs to be reset after the latest focus is applied
+    useEffect(() => {
+        !!focus && setFocus()
+    }, [focus])
```

Comment:
The focus only needs to be set once. This side effect clears the focus after the focus changed to either 'search' or 'addBlog'